### ISSUE TITLE
Help! can't configure properly the tool

### ISSUE THREAD

**Issue by ItsGoddy** [drk1wi/Modlishka#269]
I try to run the tool with locally with to modlishkia.io, I've done all the required steps such as making CA_cert and the key for it and placing it in the plugin/autocert.go also added the cert to my browser settings and when i try to run the program using:

`./dist/proxy -config templates/google.com_gsuite.json` 

the program seems to run normal but when i navigating to the phishing site it just doesn't recognize that user have accessed to it, and when i try to login using google credentials i receive "the browser isn't  secure" and wont let me enter password:
![2](https://user-images.githubusercontent.com/90037285/142464287-bc9cae89-1835-4df7-a1ac-dc71aa42657f.png)

and meanwhile the program doesn't show even that the user accessed the proxy:
![first](https://user-images.githubusercontent.com/90037285/142464386-ea69d482-b96f-4caf-a86f-8e60425f6214.png)

I'm using the this template of google.com_gsuite.json:
`{
  "proxyDomain": "loopback.modlishka.io",
  "listeningAddress": "127.0.0.1",
  "proxyAddress": "",
  "target": "https://google.com",
  "targetResources": "content.googleapis.com,www.gstatic.com,ssl.gstatic.com,ogs.google.com,accounts.google.com,clients1.google.com,clients2.google.com,clients3.google.com,clients4.google.com,clients5.google.com,clients6.google.com",
  "rules": "",
  "terminateTriggers": "",
  "terminateRedirectUrl": "",
  "trackingCookie": "ident",
  "trackingParam": "ident",
  "jsRules":"",
  "jsReflectParam": "reflect",
  "debug": false,
  "forceHTTPS": false,
  "forceHTTP": false,
  "dynamicMode": false,
  "logPostOnly": false,
  "disableSecurity": false,
  "log": "google.log",
  "plugins": "all",
  "credParams": "dHJ1ZVxdLCIoKD86XHcrW1wuXC1cX10pezAsfVx3KykiXQ==,XGJudWxsLFxbIihbYS16QS1aMC05IiEiIyQlJicoKSorLC0uLzo7PD0+P0BeX2B7fH1+XSspIixudWxsXGI=",
  "cert": "",
  "certKey": "",
  "certPool": ""
}
`

**Comment by jumny** [comment]
No communication, listening address to 0.0.0.0 which is local. No inserted certificate or key. This will not work with google anymore i would have thought so try something less secure

**Comment by ItsGoddy** [comment]
> No communication, listening address to 0.0.0.0 which is local. No inserted certificate or key. This will not work with google anymore i would have thought so try something less secure

can you give an example of a less secure option?

**Comment by jumny** [comment]
Well google/gmail now uses a system whereby if the device is not recognised you have to do a physical confirmation on a different device so even with cookies and a 2fa code it doesn't seem that you would be able to bypass, just a guess from when i've had to sign in on different devices. From a testing point of view, any other email provider or any website, it would be a case of trial and error to see who has weaknesses, that is the point of it. This software needs quiet a lot of configuring and changing to work today, it is 2 years old and unfinished, technology and fraud prevention has moved on a long way, it's beyond my skills to rewrite it but if you want some pointers the plugin/control.go needs adjustment to get the control panel to display extra information such as 2fa, apparently it should "intercept" but i can find nothing to suggest it will or string for it to read. The program works in principle but i have seen no evidence it does in the real world. Any cookies it could potentially pickup would need converting into a readable/useable format, that section states it is unfinished.

**Comment by ItsGoddy** [comment]
> Well google/gmail now uses a system whereby if the device is not recognised you have to do a physical confirmation on a different device so even with cookies and a 2fa code it doesn't seem that you would be able to bypass, just a guess from when i've had to sign in on different devices. From a testing point of view, any other email provider or any website, it would be a case of trial and error to see who has weaknesses, that is the point of it. This software needs quiet a lot of configuring and changing to work today, it is 2 years old and unfinished, technology and fraud prevention has moved on a long way, it's beyond my skills to rewrite it but if you want some pointers the plugin/control.go needs adjustment to get the control panel to display extra information such as 2fa, apparently it should "intercept" but i can find nothing to suggest it will or string for it to read. The program works in principle but i have seen no evidence it does in the real world. Any cookies it could potentially pickup would need converting into a readable/useable format, that section states it is unfinished.

thanks a lot, i will check what i can understand and maybe modify in plugin/control.go , do you have any recomendations on similar tools that are pretty much up to date? i only saw evilginx2 and didn't really tested it yet.

**Comment by jumny** [comment]
No worries, i modified the control plugin so much as to include mfa log which actually doesn't really matter as these days anyway due to time restraints and one time use. Word of warning though You'd have to be very careful using it anyway as i found all the information intercepted goes back to various github folders on this site then onto your control panel so there is some serious security issues for anybody else intending to use it, and you would have to rewrite most of it to get around this. I haven't used evilginx but might have a play with it at some point

**Comment by ItsGoddy** [comment]
> No worries, i modified the control plugin so much as to include mfa log which actually doesn't really matter as these days anyway due to time restraints and one time use. Word of warning though You'd have to be very careful using it anyway as i found all the information intercepted goes back to various github folders on this site then onto your control panel so there is some serious security issues for anybody else intending to use it, and you would have to rewrite most of it to get around this. I haven't used evilginx but might have a play with it at some point

ok thank you very much, that's pretty much sums my questions!

**Comment by danicastone** [comment]
I've found Evilginx2 to have more of the same problems. I had to rewrite a lot of it to make it use modules, and then I realized that it no longer included the google-related phishlets. Based on comments on github, people seem to think that the phishlet was pulled for being too dangerous, and will archly tell people that they should be able to rebuild it if they pay attention to how gsuite blocks javascript, or something like that.
